apps/api/src/main/java/com/agricloud/controller/AuthController.java

```diff
@@ -7,13 +7,19 @@
 
 import com.agricloud.response.GeneralResponse;
 
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import org.springframework.web.bind.annotation.GetMapping;
+
 @RestController
-@RequestMapping("/auth")
 public class AuthController {
-    
-    @PostMapping("/login")
-    public GeneralResponse login(@RequestBody String body) {
-        return new GeneralResponse("NOT IMPLEMENTED", null);
-    }
 
+    @GetMapping("/login")
+    public String handleLogin() {
+        // Redirect the user to the OAuth2 authorization endpoint for Cognito
+        String cognitoAuthorizationUrl = "localhost:5000/oauth2/authorization/cognito";
```

Comment:
Harded-coded localhost URL won't work in production (or even if the port changes).

I still get the same error message unless I'm hosting the spring application on port 5000.

Perhaps you can get the URL that the request was made from and use that to build the cognito auth URL?

Reply:
> Harded-coded localhost URL won't work in production (or even if the port changes).
> 
> I still get the same error message unless I'm hosting the spring application on port 5000.
> 
> Perhaps you can get the URL that the request was made from and use that to build the cognito auth URL?

Yeah, still busy working on it

Reply:
Then I was a bit hasty 😁